test/uk/gov/hmrc/vatsubscription/models/ReturnPeriodSpec.scala

```diff
@@ -145,34 +146,79 @@ class ReturnPeriodSpec extends UnitSpec {
 
   "ReturnPeriod .filterReturnPeriod" when {
 
-    "return period is in valid values" should {
+    "the enableAnnualAccounting feature switch is enabled" should {
 
       val returnPeriodMA: Option[ReturnPeriod] = Some(MAReturnPeriod(None))
       val returnPeriodMB: Option[ReturnPeriod] = Some(MBReturnPeriod(None))
       val returnPeriodMC: Option[ReturnPeriod] = Some(MCReturnPeriod(None))
       val returnPeriodMM: Option[ReturnPeriod] = Some(MMReturnPeriod(None))
-
-      "return the same return period back" in {
-        ReturnPeriod.filterReturnPeriod(returnPeriodMA) shouldBe returnPeriodMA
-        ReturnPeriod.filterReturnPeriod(returnPeriodMB) shouldBe returnPeriodMB
-        ReturnPeriod.filterReturnPeriod(returnPeriodMC) shouldBe returnPeriodMC
-        ReturnPeriod.filterReturnPeriod(returnPeriodMM) shouldBe returnPeriodMM
+      val returnPeriodYA: Option[ReturnPeriod] = Some(YAReturnPeriod(None))
+      val returnPeriodYB: Option[ReturnPeriod] = Some(YBReturnPeriod(None))
+      val returnPeriodYC: Option[ReturnPeriod] = Some(YCReturnPeriod(None))
+      val returnPeriodYD: Option[ReturnPeriod] = Some(YDReturnPeriod(None))
+      val returnPeriodYE: Option[ReturnPeriod] = Some(YEReturnPeriod(None))
+      val returnPeriodYF: Option[ReturnPeriod] = Some(YFReturnPeriod(None))
+      val returnPeriodYG: Option[ReturnPeriod] = Some(YGReturnPeriod(None))
+      val returnPeriodYH: Option[ReturnPeriod] = Some(YHReturnPeriod(None))
+      val returnPeriodYI: Option[ReturnPeriod] = Some(YIReturnPeriod(None))
+      val returnPeriodYJ: Option[ReturnPeriod] = Some(YJReturnPeriod(None))
+      val returnPeriodYK: Option[ReturnPeriod] = Some(YKReturnPeriod(None))
+      val returnPeriodYL: Option[ReturnPeriod] = Some(YLReturnPeriod(None))
+
+      "return the same return periods back for all returnPeriods" in {
+        ReturnPeriod.filterReturnPeriod(returnPeriodMA, mockAppConfig) shouldBe returnPeriodMA
```

Comment:
I feel like this could be written in a better way. Potentially a loop to go through a valid list. It would be easier to adjust in the future than rather having this large number of comparison statements. What do you think?